### ISSUE TITLE
Support plugins when using the markdig engine

### ISSUE THREAD

**Issue by fleed** [dotnet/docfx#3261]
[Custom markdown plugins](https://dotnet.github.io/docfx/tutorial/howto_customize_docfx_flavored_markdown.html) are only supported with the dfm markdown engine (see [this comment](
https://github.com/dotnet/docfx/issues/2837#issuecomment-415062752)).
It would be nice if also the markdig engine could support similar plugins.

**Comment by yufeih** [comment]
What markdown extensions do you need? Is it a common extension, or a really specific one that you write for your own?

Our current plan for markdig extensibility is pretty conservative, we could support a built in set of markdown extensions that the community finds useful, but we are unlikely to open an DLL injection point in the near term.

**Comment by fleed** [comment]
I’d need some generic extensions to support diagrams: something supporting “diagram as code” (mermaidjs) and something more advanced to render Visio diagrams.
I think that they would be helpful for everyone

**Comment by yufeih** [comment]
Similar to https://github.com/dotnet/docfx/issues/2347

**Comment by fleed** [comment]
@yufeih yes that issue covers the first plugin that I need. The second is still "pending". Should I close this issue and create a specific request for a Visio extension?

**Comment by yufeih** [comment]
Yes, thank you for bringing this up
